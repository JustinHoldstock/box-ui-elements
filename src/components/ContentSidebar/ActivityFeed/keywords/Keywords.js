/**
 * @flow
 * @file Keywords components
 */

import React from 'react';
import type { Node } from 'react';
import { FormattedMessage } from 'react-intl';

import Info from './Info';
import messages from '../../../messages';

import './Keywords.scss';

function getMessageForAction(action: string): Node {
    switch (action) {
        case 'applied':
            return <FormattedMessage {...messages.keywordsApplied} />;
        default:
            return null;
    }
}

type Props = {
    action: string | 'applied',
    words: string
};

const Keywords = ({ action, words }: Props): Node => (
    <div className='bcs-keywords'>
        <span className='bcs-keywords-message'>{getMessageForAction(action)}</span>
        {words ? <Info words={words} /> : null}
    </div>
);

Keywords.displayName = 'Keywords';

export default Keywords;
