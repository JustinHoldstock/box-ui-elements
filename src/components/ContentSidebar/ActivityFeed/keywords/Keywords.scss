@import '../../../styles/variables';

.box-ui-collapsed-keywords,
.box-ui-keywords {
    align-items: center;
    background-color: #f5ebfd;
    border-radius: 4px;
    color: #a64dee;
    display: flex;
    justify-content: space-between;
    margin-top: 10px;
    padding: 9px;
    width: 100%;

    .box-ui-keywords-actions {
        display: flex;
        justify-content: flex-end;
    }

    .box-ui-keywords-message {
        min-width: 0;
        overflow: hidden;
        text-overflow: ellipsis;
    }

    .box-ui-keywords-info {
        flex: none;
        position: relative;
        height: 16px;
        width: 16px;

        path {
            fill: #a64dee;
        }
    }

    .box-ui-keywords-range {
        white-space: nowrap;
    }
}

.box-ui-keywords-actions-tooltip {
    max-width: 250px;

    > span {
        display: inline-block;
        max-width: 100%;
        overflow: hidden;
        text-overflow: ellipsis;
        white-space: nowrap;
    }
}
