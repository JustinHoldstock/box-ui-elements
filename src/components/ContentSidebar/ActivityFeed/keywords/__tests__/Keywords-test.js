import React from 'react';
import { shallow } from 'enzyme';

import Keywords from '../Keywords';

describe('features/activity-feed/keywords/Keywords', () => {
    test('should correctly render keywords', () => {
        const props = {
            action: 'applied',
            words: 'cartoon font logo brand clip art illustration line artwork'
        };

        const wrapper = shallow(<Keywords {...props} />);

        expect(wrapper.find('.box-ui-keywords-message').length).toBe(1);
        expect(wrapper.find('Info').length).toBe(1);
    });

    test('should not render info icon if words are not passed', () => {
        const props = { action: 'applied', words: '' };

        const wrapper = shallow(<Keywords {...props} />);

        expect(wrapper.find('.box-ui-keywords-message').length).toBe(1);
        expect(wrapper.find('Info').length).toBe(0);
    });

    test('should correctly render info icon', () => {
        const wrapper = shallow(<Keywords action='applied' words='cartoon font logo' />);
        const info = shallow(wrapper.find('Info').getElement());

        expect(info.find('Tooltip').length).toBe(1);
        expect(info.find('IconInfoInverted').length).toBe(1);
    });
});
