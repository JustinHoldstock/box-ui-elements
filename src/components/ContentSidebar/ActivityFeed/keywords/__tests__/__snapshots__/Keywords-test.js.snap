// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features/activity-feed/keywords/Keywords should correctly render info icon 1`] = `
<span
  className="bcs-keywords-actions"
>
  <Tooltip
    className="bcs-keywords-actions-tooltip"
    constrainToScrollParent={false}
    constrainToWindow={true}
    position="bottom-left"
    text={
      <FormattedMessage
        values={
          Object {
            "words": "cartoon font logo",
          }
        }
      />
    }
    theme="default"
  >
    <div
      className="bcs-keywords-info"
    >
      <IconInfoInverted
        height={16}
        width={16}
      />
    </div>
  </Tooltip>
</span>
`;

exports[`features/activity-feed/keywords/Keywords should correctly render keywords 1`] = `
<div
  className="bcs-keywords"
>
  <span
    className="bcs-keywords-message"
  >
    <FormattedMessage />
  </span>
  <Info
    words="cartoon font logo brand clip art illustration line artwork"
  />
</div>
`;

exports[`features/activity-feed/keywords/Keywords should not render info icon if words are not passed 1`] = `
<div
  className="bcs-keywords"
>
  <span
    className="bcs-keywords-message"
  >
    <FormattedMessage />
  </span>
</div>
`;
